common/configuration/manager.go

```diff
@@ -84,9 +84,8 @@ func (m *Manager[R]) getData(ctx context.Context, ref Ref) ([]byte, error) {
 	key, err := m.resolver.Get(ctx, ref)
 	// Try again at the global scope if the value is not found in module scope.
 	if ref.Module.Ok() && errors.Is(err, ErrNotFound) {
-		gref := ref
-		gref.Module = optional.None[string]()
-		key, err = m.resolver.Get(ctx, gref)
+		ref.Module = optional.None[string]()
```

Comment:
If global config/secret is found, we need the `ref` var to point to it as it is used to access it from the provider.
Previously, resolver would match the global ref, but we would then use the module specific ref with the provider.